app/src/main/java/com/github/factotum_sdp/factotum/ui/directory/ContactDetailsFragment.kt

```diff
@@ -23,22 +24,40 @@ class ContactDetailsFragment : Fragment() {
         val button = mainView.findViewById<Button>(R.id.button) // connect the button to the layout
         button.setOnClickListener { view ->
             view.findNavController().navigate(R.id.action_contactDetailsFragment2_to_directoryFragment)} // go back to the list of contacts when the button is clicked
-        setContactDetails(mainView, ContactsList.getItems()[arguments?.getInt("id")!!]) //links the contact details to the layout
+        val contactsViewModel =
+            ViewModelProvider(requireActivity())[ContactsViewModel::class.java] // get the contacts view model
+        setContactDetails(mainView,
+            contactsViewModel.contacts.value?.get(arguments?.getInt("id")!!)!!
```

Comment:
You could use the safe call operator ?. followed by the elvis operator ?: to provide a default value.